### DISCUSSION TITLE
lesson 9 problem in interactions.s.sol

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#634] [General]
i am not able to understand these function how parameters are passed in them how they are working.
```
  function CreateSubscriptionUsingConfig() public returns (uint64) {
        HelperConfig helperConfig = new HelperConfig();
        (,, address vrfCoordinator,,,) = helperConfig.activeNetworkConfig();
        return createSubscription(vrfCoordinator);
    }

    function createSubscription(address vrfCoordinator) public returns (uint64) {
        console.log("creating SubscriptionId on chainId:", block.chainid);
        vm.startBroadcast();
        uint64 subId = VRFCoordinatorV2Mock(vrfCoordinator).createSubscription();
        vm.stopBroadcast();
        console.log("your subscriptionId is :", subId);
       return subId;
```

**Comment by Abdulllah-Rizwan** [comment]
you are not the only one who got confused while working on the lottery project.

**Reply by khushbusandhu** [reply]
Maybe but I got very much confused

**Reply by theirrationalone** [reply]
It will be a So long description if I start talking about its working and the concept behind it.... Better to use `chatGpt` (it may give wrong explanation but it's so elegant in coding) Please refer to `chatGpt` with same Question.

Use `shift + Enter` key chord to shift to new line inside text input box and also you backticks `\`` too, to describe code.

**Reply by khushbusandhu** [reply]
yes, I did the same.
I searched on Phind about how these functions are working and I find it very much helpful

**Reply by theirrationalone** [reply]
Wooaahhh 👏👏, It will be a Great Plus if You become Skilled in How to unleash The power of AI Tools.

**Reply by khushbusandhu** [reply]
hey @theirrationalone! can I contact you with doubts about the foundry and other web3 concepts?

**Reply by theirrationalone** [reply]
Heyy @khushbusandhu !, I'm so Sorry for the late reply since i was busy 😅, 

My twitter [theirrationalon](https://x.com/theirrationalon?t=WboOdxHPMVZ5wp3pDzdnQQ&s=09)
